src/modules/launcher/Plugins/Microsoft.Plugin.Calculator.UnitTest/ExtendedCalculatorParserTests.cs

```diff
@@ -120,6 +121,10 @@ public void Interpret_DifferentCulture_WhenCalled(string input, decimal expected
         [TestCase("sqrt( 36)", true)]
         [TestCase("max 4", false)]
         [TestCase("sin(0)", true)]
+        [TestCase("sinh(1)", true)]
+        [TestCase("tanh(0)", true)]
+        [TestCase("artanh(pi/2)", true)]
+        [TestCase("cosh", false)]
```

Comment:
Let's make sure to have tests checking that nothing is returned when arcosh and artanh are called outside their domain.
The behavior I'm seeing is correct.

Reply:
What do you mean by outside their domain?
Add testcases `[TestCase("arcosh", false)]` and `[TestCase("artanh", false)]`?

Reply:
arcosh(x) is only defined for x >= 1 and artanh(x) only for -1 < x < 1.
So we could add, say arcosh(x) for x in {-1, 0, 0.99} and artanh(x) for x in {-1, 1, 10}.
These should all return false.

Interestingly, tanh(artanh(1)) evaluates to 1, probably because the intermediate result is +infinity and tanh can deal with that.

Reply:
Ah okay!
But then I'll also add tests for `arcsin(5)`, `log(-1)` and `log(0)` (and for `ln`).

Reply:
Sure, why not!

Reply:
At the moment, those cases are completely ignored and exceptions are thrown. 
If the result from mages is `NaN`, it gets transformed into a string `Not a number (NaN)` and then the conversion to decimal fails. Same for `Infinity` and `-Infinity`, because `decimal` doesn't support those numbers.
I could just add a check if it is of type string and then return this string. Then `TransformResult` in `CalculateEngine` would make sense.